builtin/fetch.c

```diff
@@ -958,8 +958,10 @@ static int store_updated_refs(const char *raw_url, const char *remote_name,
 				ref->force = rm->peer_ref->force;
 			}
 
-			if (recurse_submodules != RECURSE_SUBMODULES_OFF)
+			if (recurse_submodules != RECURSE_SUBMODULES_OFF && !oideq(
+						&ref->old_oid, &ref->new_oid)) {
```

Comment:
The common way to wrap things like this in Git is to wrap directly after the `&&` and then align the `!` just under the first character after the opening `(`. I.e.
```suggestion
			if (recurse_submodules != RECURSE_SUBMODULES_OFF &&
			    !oideq(&ref->old_oid, &ref->new_oid)) {
```